mlrun/runtimes/base.py

```diff
@@ -1148,42 +1311,62 @@ def _ensure_runtime_resource_run_logs_collected(
                 store_log = True
 
         if store_log:
-            logger.info("Storing runtime resource log before deletion")
-            logs_from_k8s, _ = crud.Logs.get_log(
+            _, logs_from_k8s = crud.Logs.get_logs(
                 db_session, project, uid, source=LogSources.K8S
             )
-            crud.Logs.store_log(logs_from_k8s, project, uid, append=False)
+            if logs_from_k8s:
+                logger.info("Storing finished run logs", project=project, uid=uid)
+                crud.Logs.store_log(logs_from_k8s, project, uid, append=False)
 
     @staticmethod
-    def _ensure_runtime_resource_run_status_updated(
+    def _ensure_run_state(
         db: DBInterface,
         db_session: Session,
         project: str,
         uid: str,
         desired_run_state: str,
```

Comment:
I explained here - https://github.com/mlrun/mlrun/pull/467#discussion_r502185292 why there might be two different terminal states
anyways:
1. changed to `run_state` and `db_run_state`
2. changed so that `run_state` will take precedence over `db_run_state` always

Reply:
Answered in original thread: https://github.com/mlrun/mlrun/pull/467#discussion_r504947101
Resolving this one to reduce noise